### PR TITLE
Make ephemeral storage customizable for tasks

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Make ephemeral storage customizable for tasks. This is now broadly supported on kubernetes.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/494

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/common/tasks/raw_container.py [hunk 1]
```diff
@@ -62,6 +72,12 @@ def _get_container_definition(
     limits = []
     if storage_limit:
         limits.append(_task_models.Resources.ResourceEntry(_task_models.Resources.ResourceName.STORAGE, storage_limit))
+    if ephemeral_storage_limit:
+        limits.append(
```
Comment (cosmicBboy) [inline flytekit/common/tasks/raw_container.py:76]: update tests to cover this case?

Comment (katrogan) [reply flytekit/common/tasks/raw_container.py:76]: @cosmicBboy done

**Review by cosmicBboy — DISMISSED** [review]
✨

**Comment by katrogan** [comment]
had to rebase, mind re-approving @cosmicBboy

**Review by cosmicBboy — APPROVED** [review]

**Pull request merged**